### PR TITLE
Profile page

### PR DESCRIPTION
**Pull request by mikesch803**
1. add profile page
2. add favicon
3. add search feature
4. refactor form validation code

### CODE REVIEW

File: src/components/header/Header.jsx [hunk 1]
```diff
@@ -1,29 +1,39 @@
...
+  const searchVideoHandler = e => {
+        navigate('/trend');
+        dispatch({type:"SEARCH_VIDEO", payload: e.target.value})
+  }
+
+  // function debounce(fn, delay){
+  //   let timer;
+  //   return function(){
+  //     clearTimeout(timer);
+  //     timer = setTimeout(()=> fn(), delay);
```
Comment (nayyyhaa) [inline src/components/header/Header.jsx]: can remove commented code if not required.

File: src/context/video-context.jsx [hunk 2]
```diff
@@ -35,6 +39,11 @@ const VideoProvider = ({ children }) => {
     })();
   }, []);
 
+  // const searchVideoHandler = (search) => {
+  //   navigate('/trend');
+  //   allVideos.filter(item => item.title.toLowerCase().match(search.toLowerCase()))
+  // }
+
```
Comment (nayyyhaa) [inline src/context/video-context.jsx:46]: remove code from here too.

File: src/hooks/useTitle.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { useEffect } from 'react'
+
+export const useTitle = (page) => {
+    useEffect(()=>{
+        document.title = `FitVideo | ${page}`
```
Comment (nayyyhaa) [inline src/hooks/useTitle.js:5]: appreciation: nice way of giving dynamic page titles

File: src/pages/history-page/History.jsx [hunk 4]
```diff
@@ -43,7 +44,11 @@ export function History() {
         <div className="video-container">
           {historyVideos.map((item) => (
             <li key={item._id} className="history-list">
-              <VideoCard item={item} videoCardOptionState={videoCardOptionState} setVideoCardOptionState={setVideoCardOptionState}/>
+              <VideoCard
+                item={item}
+                videoCardOptionState={videoCardOptionState}
+                setVideoCardOptionState={setVideoCardOptionState}
```
Comment (nayyyhaa) [inline src/pages/history-page/History.jsx:50]: appreciation: good that you are sending specific state of video via props.

File: src/utils/functions/filterVideos.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+export const filterVideos = (state) => {
+  let sortedVideos = [...state.allVideos];
+  if(state.searchVideos) {
+      console.log(state.searchVideos)
```
Comment (nayyyhaa) [inline src/utils/functions/filterVideos.js:4]: remove console from here as well..

**Review by nayyyhaa — APPROVED** [review]
Nice work. only a few console logs are present which you can remove. Rest you can merge the changes.

**Review by Dev-AkshatGupta — APPROVED** [review]
Very small pr just consider the suggestions given by the prior reviewer and you are good to go after wards. Great work.

**Pull request merged**